src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/ImageSequenceReference.java

```diff
@@ -0,0 +1,214 @@
+            int frameStep,
+            double rate,
+            int frameZeroPadding,
+            MissingFramePolicy missingFramePolicy,
+            TimeRange availableRange,
+            AnyDictionary metadata) {
+        this.initObject(
+                targetURLBase,
+                namePrefix,
+                nameSuffix,
+                startFrame,
+                frameStep,
+                rate,
+                frameZeroPadding,
+                missingFramePolicy,
+                availableRange,
+                metadata);
+    }
+
+    public ImageSequenceReference(ImageSequenceReference.ImageSequenceReferenceBuilder imageSequenceReferenceBuilder) {
+        this.initObject(
+                imageSequenceReferenceBuilder.targetURLBase,
+                imageSequenceReferenceBuilder.namePrefix,
+                imageSequenceReferenceBuilder.nameSuffix,
+                imageSequenceReferenceBuilder.startFrame,
+                imageSequenceReferenceBuilder.frameStep,
+                imageSequenceReferenceBuilder.rate,
+                imageSequenceReferenceBuilder.frameZeroPadding,
+                imageSequenceReferenceBuilder.missingFramePolicy,
+                imageSequenceReferenceBuilder.availableRange,
+                imageSequenceReferenceBuilder.metadata);
+    }
+
+    private void initObject(String targetURLBase,
+                            String namePrefix,
+                            String nameSuffix,
+                            int startFrame,
+                            int frameStep,
+                            double rate,
+                            int frameZeroPadding,
+                            MissingFramePolicy missingFramePolicy,
+                            TimeRange availableRange,
+                            AnyDictionary metadata) {
+        this.className = this.getClass().getCanonicalName();
+        this.initialize(
+                targetURLBase,
+                namePrefix,
+                nameSuffix,
+                startFrame,
+                frameStep,
+                rate,
+                frameZeroPadding,
+                missingFramePolicy.ordinal(),
+                availableRange,
+                metadata);
+    }
+
+    private native void initialize(String targetURLBase,
+                                   String namePrefix,
+                                   String nameSuffix,
+                                   int startFrame,
+                                   int frameStep,
+                                   double rate,
+                                   int frameZeroPadding,
+                                   int missingFramePolicy,
+                                   TimeRange availableRange,
+                                   AnyDictionary metadata);
+
+    public static class ImageSequenceReferenceBuilder {
+        String targetURLBase = "";
+        String namePrefix = "";
+        String nameSuffix = "";
+        int startFrame = 1;
```

Comment:
do frames start with 0 or 1? @reinecke - any idea?

Reply:
It depends on what was chosen by the creator of the image sequence, but the common conventions for start frame tend to be 1, 101, or 1001.

Reply:
I've used this initial value as per C++ implementation.

Reply:
Awesome thank you all, I am just using this opportunity to learn :)